### PR TITLE
DOCS-4512 environment hardening

### PR DESCRIPTION
**Pull request by paigehargrave**

### CODE REVIEW

File: modules/ROOT/pages/configure-hardening.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+= Configure Environment Hardening on Runtime Fabric
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+By design, Runtime Fabric automatically provides multiple aspects of environment hardening for your Mule installation.
```
Comment (hannanelson) [inline modules/ROOT/pages/configure-hardening.adoc]: methods? (aspects sounds weird)

File: modules/ROOT/pages/configure-hardening.adoc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+The following hardening checklist provides additional information about environment hardening already provided by 
+Runtime Fabric, and specifies Runtime Fabric coverage as well as areas that require additional configuration.
+
+== Hardening Checklist
+
+[%header%autowidth.spread]
+|===
+| Hardening Step | Action Required
+| Run Mule as a Non-privileged User. | RTF provides this functionality.
```
Comment (hannanelson) [inline modules/ROOT/pages/configure-hardening.adoc]: non

Comment (hannanelson) [inline modules/ROOT/pages/configure-hardening.adoc]: Kim's terminology list says no to RTF (global).

File: modules/ROOT/pages/configure-hardening.adoc [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+| Install Mule as a Service |	RTF provides this functionality by deploying a Mule runtime for each application within its own container.
+| Configure Mule to write logs or temporary files within appropriate locations. Configure logs, passwords, and keystore files. | RTF provides this functionality.  Mule writes to a temporary file system that is deleted when the application terminates. Monitoring information is collected from a sidecar agent in a specific location on disk where metrics are outputted and stored. Logs are collected from standard out and standard error streams.
```
Comment (hannanelson) [inline modules/ROOT/pages/configure-hardening.adoc]: output?

Comment (hannanelson) [reply modules/ROOT/pages/configure-hardening.adoc]: Check extra spaces (double space after period).

File: modules/ROOT/pages/configure-hardening.adoc [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+| In some situations, you must configure usernames, passwords, and keystores on Mule. Usually, these settings are made available externally, so that dev/ops can change these settings. |	RTF provides this functionality. Refer to xref:manage-secure-properties.adoc[Manage Secure Properties in Runtime Fabric] for additional information.
```
Comment (hannanelson) [inline modules/ROOT/pages/configure-hardening.adoc]: Is this usually how this is presented? I always see DevOps.

File: modules/ROOT/pages/configure-hardening.adoc [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+| Manage certificates in a keystore file. | This is supported on RTF, but you are responsible for importing the certificates into the keystore. In addition, RTF includes a load balancer which can handle SSL termination for all Mules deployed within it.
```
Comment (hannanelson) [inline modules/ROOT/pages/configure-hardening.adoc]: must import?

Comment (hannanelson) [inline modules/ROOT/pages/configure-hardening.adoc]: I think "Mules" is verboten. "Mule apps"?
Check Terminology doc: https://docs.google.com/document/d/1irvvVsV4fmJoA7a14cmINrVna8TeFcsBKM76JXhKkIM/edit

Comment (hannanelson) [reply modules/ROOT/pages/configure-hardening.adoc]: , which?

**Review by hannanelson — APPROVED** [review]
Some suggestions :)

File: modules/ROOT/pages/configure-hardening.adoc (hunk 1 above)
Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: Fixed - thanks!

File: modules/ROOT/pages/configure-hardening.adoc (hunk 2 above)
Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: got it - thank you!

Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: Thank you!

File: modules/ROOT/pages/configure-hardening.adoc (hunk 3 above)
Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: Thank you!

File: modules/ROOT/pages/configure-hardening.adoc (hunk 4 above)
Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: Yup - thank you!

File: modules/ROOT/pages/configure-hardening.adoc (hunk 5 above)
Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: Good catches - thank you!

Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: Fixed - thank you!

**Comment by paigehargrave** [comment]
@hannanelson Thank you for review!

File: modules/ROOT/nav.adoc [hunk 6]
```diff
@@ -12,6 +12,7 @@
 * xref:runtime-fabric-logs.adoc[View and Configure Logs in Runtime Fabric]
 * xref:configure-alerting.adoc[Configure Alerting on Anypoint Runtime Fabric]
 * xref:deploy-index.adoc[Deploy Applications and API Proxies to Runtime Fabric]
+ ** xref:harden-env.adoc[Perform Environment Hardening]
```
Comment (sundippatel) [inline modules/ROOT/nav.adoc]: I think "Hardening Mule" makes more sense.

File: modules/ROOT/pages/configure-hardening.adoc [hunk 7]
```diff
@@ -0,0 +1,23 @@
+= Configure Environment Hardening on Runtime Fabric
```
Comment (sundippatel) [inline modules/ROOT/pages/configure-hardening.adoc]: Hardening Mule on Runtime Fabric

File: modules/ROOT/pages/configure-hardening.adoc (hunk 1 above)
Comment (sundippatel) [inline modules/ROOT/pages/configure-hardening.adoc]: "..automatically applies and enables hardening measures for your Mule applications."

Something closer to that I think makes more sense to what the user is after.

File: modules/ROOT/pages/configure-hardening.adoc [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+By design, Runtime Fabric automatically provides multiple aspects of environment hardening for your Mule installation. 
+For example, by leveraging Docker and Kubernetes, app isolation is achieved by default.
```
Comment (sundippatel) [inline modules/ROOT/pages/configure-hardening.adoc]: and by following containerization best practices, isolation at the Mule runtime and application level is achieved by default.

File: modules/ROOT/pages/configure-hardening.adoc [hunk 9]
```diff
@@ -0,0 +1,23 @@
+= Configure Environment Hardening on Runtime Fabric
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+By design, Runtime Fabric automatically provides multiple aspects of environment hardening for your Mule installation. 
+For example, by leveraging Docker and Kubernetes, app isolation is achieved by default.  
+
+The following hardening checklist provides additional information about environment hardening already provided by
```
Comment (sundippatel) [inline modules/ROOT/pages/configure-hardening.adoc]: unclear on what "environment hardening" means. Sub with "Mule hardening"

File: modules/ROOT/pages/configure-hardening.adoc [hunk 10]
```diff
@@ -0,0 +1,23 @@
+= Configure Environment Hardening on Runtime Fabric
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+By design, Runtime Fabric automatically provides multiple aspects of environment hardening for your Mule installation. 
+For example, by leveraging Docker and Kubernetes, app isolation is achieved by default.  
+
+The following hardening checklist provides additional information about environment hardening already provided by 
+Runtime Fabric, and specifies Runtime Fabric coverage as well as areas that require additional configuration.
```
Comment (sundippatel) [inline modules/ROOT/pages/configure-hardening.adoc]: and specifies coverage provided by default as well as how additional methods that can be applied with Runtime Fabric.

File: modules/ROOT/pages/configure-hardening.adoc (hunk 2 above)
Comment (sundippatel) [reply modules/ROOT/pages/configure-hardening.adoc]: Unsure if we've introduced the term "RTF" in our docs (some might associate it with "rich text format"). 

Ask to check in order to be consistent.

Comment (sundippatel) [reply modules/ROOT/pages/configure-hardening.adoc]: "Runtime Fabric runs Mule applications as a non-privileged user by default.:

File: modules/ROOT/pages/configure-hardening.adoc (hunk 3 above)
Comment (sundippatel) [reply modules/ROOT/pages/configure-hardening.adoc]: ..are collected from standard output streams. -- i think that'd be sufficient.

**Review by sundippatel — CHANGES_REQUESTED** [review]
Good stuff. Left some suggestions.

File: modules/ROOT/nav.adoc (hunk 6 above)
Comment (paigehargrave) [reply modules/ROOT/nav.adoc]: Updated - thank you!

File: modules/ROOT/pages/configure-hardening.adoc (hunk 7 above)
Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: Updated - thank you!

File: modules/ROOT/pages/configure-hardening.adoc (hunk 1 above)
Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: OK - got it - thank you!

File: modules/ROOT/pages/configure-hardening.adoc (hunk 8 above)
Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: Got it - please make sure I added what was intended - thank you!

File: modules/ROOT/pages/configure-hardening.adoc (hunk 9 above)
Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: Got it - thanks!

File: modules/ROOT/pages/configure-hardening.adoc (hunk 3 above)
Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: Got it - thanks!

**Review by sundippatel — APPROVED** [review]

File: modules/ROOT/pages/configure-hardening.adoc (hunk 10 above)
Comment (paigehargrave) [reply modules/ROOT/pages/configure-hardening.adoc]: This got reworded.

**Pull request merged**